sympy/combinatorics/pc_groups.py

```diff
@@ -0,0 +1,494 @@
+        >>> for s, g in zip(group.symbols, pcgs):
+        ...     free_to_perm[s] = g
+
+        >>> for k, v in pc_resentation.items():
+        ...     k_array = k.array_form
+        ...     if v != ():
+        ...        v_array = v.array_form
+        ...     lhs = Permutation()
+        ...     for gen in k_array:
+        ...         s = gen[0]
+        ...         e = gen[1]
+        ...         lhs = lhs*free_to_perm[s]**e
+        ...     if v == ():
+        ...         assert lhs.is_identity
+        ...         continue
+        ...     rhs = Permutation()
+        ...     for gen in v_array:
+        ...         s = gen[0]
+        ...         e = gen[1]
+        ...         rhs = rhs*free_to_perm[s]**e
+        ...     assert lhs == rhs
+
+        """
+        rel_order = self.relative_order()
+        pc_relators = {}
+        perm_to_free = {}
+        pcgs = self.pcgs
+
+        for gen, s in zip(pcgs, group.generators):
+            perm_to_free[gen**-1] = s**-1
+            perm_to_free[gen] = s
+
+        collector = Collector(pc_relators, rel_order, group)
+        pcgs.reverse()
+        series = self.pc_series
+        series.reverse()
+        collected_gens = []
+
+        for i, gen in enumerate(pcgs):
+            re = rel_order[len(rel_order)-i-1]
+            relation = perm_to_free[gen]**re
+            G = series[i] if i != 0 else series[i+1]
+
+            l = G.generator_product(gen**re, original = True)
+            l.reverse()
+
+            word = group.identity
+            for g in l:
+                word = word*perm_to_free[g]
+
+            word = collector.collected_word(word)
+            pc_relators[relation] = word if word else ()
+            collector.pc_relators = pc_relators
+
+            collected_gens.append(gen)
+            if len(collected_gens) > 1:
+                conj = collected_gens[len(collected_gens)-1]
+                conjecture = perm_to_free[conj]
+
+                for j in range(len(collected_gens)-1):
+                    conjugated = perm_to_free[collected_gens[j]]
+
+                    relation = conjecture**-1*conjugated*conjecture
+                    gens = conj**-1*collected_gens[j]*conj
+
+                    l = G.generator_product(gens, original = True)
+                    l.reverse()
+                    word = group.identity
+                    for g in l:
+                        word = word*perm_to_free[g]
+
+                    word = collector.collected_word(word)
+                    pc_relators[relation] = word if word else ()
+                    collector.pc_relators = pc_relators
+
+        series.reverse()
+        pcgs.reverse()
+        return pc_relators
+
+
+class Collector(DefaultPrinting):
+
+    """
+    References
+    ==========
+
+    .. [1] Holt, D., Eick, B., O'Brien, E.
+           "Handbook of Computational Group Theory"
+           Section 8.1.3
+    """
+
+    def __init__(self, pc_relators, relative_order, group):
```

Comment:
Maybe we could have `group=None` as optional parameter. Then there could be something like

    if group is None:
        group = FreeGroup(<some symbols>)

It the generators are needed somewhere, they could also be added to Collector:

    self.gens = group.generators

but maybe that is not necessary.

Reply:
Isn't if we are providing `pc_relators` then `group` must be provided?

Reply:
May be this can be done with the class `PolycyclicGroup`, then the `group` parameter can be omitted.

Reply:
> May be this can be done with the class `PolycyclicGroup`

But that also I think will not be a good choice!

Reply:
It looks like there are several possibilities. If pc_relators are given (as they are now) then the group can be found from those relators. (They contain elements of the group.) Alternatively, and maybe preferably, pcgs and relative_order could be given as parameters, and then the pc_relators could be derived from them by internal methods of Collector class. (Most of the code is there already, only differently organized.)

Reply:
>pc_relators could be derived from them by internal methods of Collector class. 
If not given by user then there is only one way to get the whole pc_relators via pc_presentation. Should there be an additional method to Collector to compute pc_relators given user provides group is a parameter to Collector.
Does it makes sense having pcgs as a parameter to Collector(because it should be provided by user) better will be take the group directly.

Reply:
I think that `pc_presentation` should be a collector method. That is where it logically belongs. Then a collector would create the presentation internally. A polycyclic group would receive a ready made collector including the presentation as an attribute. A collector can be created from a pcgs alone. A polycyclic series can be derived from it and relative orders as well (though it may sometimes be more efficient to pass them also as parameters if they have been precomputed).